### PR TITLE
✨ 实现沸点页内容的展开/收起 功能

### PR DESCRIPTION
**Pull request by shirne**

### CODE REVIEW

File: lib/pages/home/pins.dart [hunk 1]
```diff
@@ -243,3 +226,66 @@ class _PinItemWidget extends StatelessWidget {
     );
   }
 }
+
+class ReadMoreWrapper extends StatefulWidget {
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Maybe `_PinContentWidget`.

File: lib/pages/home/pins.dart [hunk 2]
```diff
@@ -243,3 +226,66 @@ class _PinItemWidget extends StatelessWidget {
...
+  const ReadMoreWrapper(this.content, {Key? key}) : super(key: key);
+  final String content;
```
Comment (AlexV525) [inline lib/pages/home/pins.dart:232]: Add an empty line before this.

File: lib/pages/home/pins.dart [hunk 3]
```diff
@@ -243,3 +226,66 @@ class _PinItemWidget extends StatelessWidget {
...
+
+  @override
+  State<ReadMoreWrapper> createState() => _ReadMoreWrapperState();
+}
+
+class _ReadMoreWrapperState extends State<ReadMoreWrapper> {
+  bool readAll = false;
+
+  _moreTap() {
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Always specify the return type.

File: lib/pages/home/pins.dart [hunk 4]
```diff
@@ -243,3 +226,66 @@ class _PinItemWidget extends StatelessWidget {
...
+class ReadMoreWrapper extends StatefulWidget {
+  const ReadMoreWrapper(this.content, {Key? key}) : super(key: key);
+  final String content;
+
+  @override
+  State<ReadMoreWrapper> createState() => _ReadMoreWrapperState();
+}
+
+class _ReadMoreWrapperState extends State<ReadMoreWrapper> {
+  bool readAll = false;
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Maybe `isExpanding`.

File: lib/pages/home/pins.dart [hunk 5]
```diff
@@ -243,3 +226,66 @@ class _PinItemWidget extends StatelessWidget {
...
+    setState(() {
+      readAll = !readAll;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
```
Comment (AlexV525) [inline lib/pages/home/pins.dart:251]: Could you explain more about why you use stretch here?

File: lib/pages/home/pins.dart [hunk 6]
```diff
@@ -243,3 +226,66 @@ class _PinItemWidget extends StatelessWidget {
...
+      children: [
+        ExtendedText(
+          widget.content,
+          maxLines: readAll ? 100 : 3,
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Why it's limited to 100?

File: lib/pages/home/pins.dart (hunk 2 above)
Comment (shirne) [reply lib/pages/home/pins.dart:232]: Two lines before class or an empty line before constructor ?
This is format by dart.

File: lib/pages/home/pins.dart (hunk 5 above)
Comment (shirne) [reply lib/pages/home/pins.dart:251]: The default value center will center the fold button(text).
The diffence between start and stretch is the `more` button 
stretch will always align more button to right.
start will align more button close to the right of the longest line of text.
微信截图_20220801010244

Comment (shirne) [reply lib/pages/home/pins.dart:251]: 微信截图_20220801010454

File: lib/pages/home/pins.dart [hunk 7]
```diff
@@ -227,20 +225,21 @@ class _PinItemWidget extends StatelessWidget {
   }
 }
 
-class ReadMoreWrapper extends StatefulWidget {
-  const ReadMoreWrapper(this.content, {Key? key}) : super(key: key);
+/// Pins content widget to fold or expand the content
+class _PinContentWidget extends StatefulWidget {
+  const _PinContentWidget(this.content, {Key? key}) : super(key: key);
   final String content;
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]:
```suggestion
/// Pins content widget to control expanding and folding.
class _PinContentWidget extends StatefulWidget {
  const _PinContentWidget(this.content, {Key? key}) : super(key: key);

  final String content;
```

File: lib/pages/home/pins.dart (hunk 5 above)
Comment (AlexV525) [reply lib/pages/home/pins.dart:251]: I don't see the relation between your screenshot.

File: lib/pages/home/pins.dart (hunk 6 above)
Comment (shirne) [reply lib/pages/home/pins.dart]: I've change it to null.

File: lib/pages/home/pins.dart (hunk 5 above)
Comment (AlexV525) [reply lib/pages/home/pins.dart:251]: Thank! That's new to know too.

**Review by AlexV525 — APPROVED** [review]
LGTM

**Pull request merged**